### PR TITLE
Moving GUI out of the controllers

### PR DESCRIPTION
**Pull request by mrvisscher**
Big refactor that moves GUI interaction out of the controllers and into specialized `ABAction`'s. This introduces a new design philosophy in which widgets are only responsible for their own state, controllers are only responsible for Brightway's state, and actions are responsible for the logic for moving between states. This:

- Separates the UI from the controllers, making them initializable before UI init, preventing circular imports
- Moves specialized user interaction from widgets, clearing up where user interaction happens
- Makes user interactions easily testable by unit-testing ABActions (>90% coverage right now)
- ABActions inherit QActions, so they can easily be added to buttons, or (context)menus.

## ABAction flowchart
ABAction flowchart

## Checklist

- [x] Update the documentation, please follow the [numpy style guide](https://numpydoc.readthedocs.io/en/latest/format.html).
- [x] Update tests.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `feature`, `ui`, `change`, `documentation`, `breaking`, `ci`
      as they show up in the changelog.
- [x] Add a milestone to the PR for the intended release.
- [x] Request a review from another developer.

### CODE REVIEW

File: activity_browser/layouts/tabs/project_manager.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
     ActivitiesBiosphereTable,
 )
 from ...signals import signals
+from ...actions import DatabaseImport, DatabaseNew, DefaultInstall, ProjectNew, ProjectDuplicate, ProjectDelete
```
Comment (marc-vdm) [inline activity_browser/layouts/tabs/project_manager.py:13]: Update all of these imports to to ab.actions

File: activity_browser/ui/tables/delegates/formula.py [hunk 2]
```diff
@@ -90,12 +90,10 @@ def __init__(self, parent=None, flags=QtCore.Qt.Window):
...
-        self.new_parameter = QtWidgets.QPushButton(
-            qicons.add, "New parameter", self
-        )
-        self.new_parameter.clicked.connect(
-            lambda: signals.add_parameter.emit(self.key)
-        )
+
+        self.new_parameter_button = QtWidgets.QToolButton(self)
+        self.new_parameter_button.setToolButtonStyle(QtCore.Qt.ToolButtonTextBesideIcon)
+        self.new_parameter_button.setDefaultAction(ParameterNew(self.get_key, self))
```
Comment (mrvisscher) [inline activity_browser/ui/tables/delegates/formula.py:96]: Switch to `get_button()` convenvience function

File: activity_browser/ui/menu_bar.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 from ..info import __version__ as ab_version
 from .icons import qicons
 from ..signals import signals
+from ..actions import *
```
Comment (marc-vdm) [inline activity_browser/ui/menu_bar.py:9]: In these cases just `import actions` and in code do `action.x`

File: activity_browser/ui/tables/models/parameters.py [hunk 4]
```diff
@@ -223,7 +233,7 @@ def sync(self) -> None:
 
     @classmethod
     def parse_parameter(cls, parameter) -> dict:
-        """ Override the base method to add more steps.
+        """ Override the base.py method to add more steps.
```
Comment (marc-vdm) [inline activity_browser/ui/tables/models/parameters.py:236]: .

File: activity_browser/ui/tables/views.py [hunk 5]
```diff
@@ -129,7 +129,7 @@ def keyPressEvent(self, e):
 
 
 class ABFilterableDataFrameView(ABDataFrameView):
-    """ Filterable base class for showing pandas dataframe objects as tables.
+    """ Filterable base.py class for showing pandas dataframe objects as tables.
```
Comment (marc-vdm) [inline activity_browser/ui/tables/views.py:132]: .

File: activity_browser/ui/widgets/line_edit.py [hunk 6]
```diff
@@ -95,6 +95,6 @@ def focusOutEvent(self, event):
         super(SignalledComboEdit, self).focusOutEvent(event)
 
     # def showPopup(self):
```
Comment (marc-vdm) [inline activity_browser/ui/widgets/line_edit.py:97]: remove?

File: tests/legacy/test_settings.py [hunk 7]
```diff
@@ -29,7 +29,7 @@ def project_settings(qtbot, ab_app):
 
 
 def test_base_class():
-    """ Test that the base class raises an error on initialization
+    """ Test that the base.py class raises an error on initialization
```
Comment (marc-vdm) [inline tests/legacy/test_settings.py:32]: .

File: tests/wizards/test_import_wizard.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+import brightway2 as bw
+from PySide2 import QtCore, QtWidgets
+
+from activity_browser.signals import signals
+from activity_browser.controllers.database import DatabaseController
+from activity_browser.ui.wizards.db_import_wizard import DatabaseImportWizard
+
+#
+#
+# def test_open_db_wizard_button(qtbot, ab_app, monkeypatch):
```
Comment (marc-vdm) [inline tests/wizards/test_import_wizard.py:11]: remove?

File: activity_browser/settings.py [hunk 9]
```diff
@@ -177,7 +177,7 @@ def get_default_project_name() -> Optional[str]:
 class ProjectSettings(BaseSettings):
     """
     Handles user settings which are specific to projects. Created initially to handle read-only/writable database status
-    Code based on ABSettings class, if more different types of settings are needed, could inherit from a base class
+    Code based on ABSettings class, if more different types of settings are needed, could inherit from a base.py class
```
Comment (marc-vdm) [inline activity_browser/settings.py:180]: .

**Review by marc-vdm — COMMENTED** [review]
Also some documentation TODO:
- [ ] Make contributing.md a smaller file with references the below:
- [ ] Add files explaining important items (file for actions, controllers, signals etc)
    - Why do they exist?
    - When (not) to use them?
    - Relationships between other things
- [ ] File on how to update tests

**Pull request merged**